### PR TITLE
Create Your-vouchers.md

### PR DESCRIPTION
**Pull request by MatShir**
All the cases that display the voucher on the voucher page

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | There is confusion if the voucher should be displayed or not. The spec describes the full case and the different parameters.
| Fixed ticket? | Fixes #https://github.com/PrestaShop/PrestaShop/issues/22397

### CODE REVIEW

File: front-office/Your-vouchers.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+## Rules of the displayed vouchers
+
+A voucher is displayed if it follows specific restrictions:
+
+On the table below are listed all the different cases.
+
+| Cases | Code | Highlighted | Group restriction | User resctriction | Is Visible Voucher page ? |
```
Comment (LouiseBonnard) [inline front-office/Your-vouchers.md:7]:
```suggestion
| Cases | Code | Highlighted | Group restriction | User restriction | Available in the FO vouchers page? |
```

File: front-office/Your-vouchers.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+|-------|------|-------------|-------------------|-------------------|---------------------------|
+| 1     | X    | Yes         | X                 | X                 | Yes                       |
+| 2     | X    | X           | X                 | Yes               | Yes                       |
+| 3     | Yes  | No          | No                | No                | No                        |
+| 4     | No   | No          | Yes               | No                | Yes                       |
+| 5     | No   | No          | No                | No                | Yes                       |
+| 6     | Yes  | No          | Yes               | No                | No                        |
+
+When there is no voucher for the customer, the page displays: "You do not have any vouchers."
```
Comment (LouiseBonnard) [inline front-office/Your-vouchers.md:16]:
```suggestion
When there is no voucher for the customer, the page displays a notification: `You do not have any vouchers.` in `Shop.Notifications.Warning`.
```

File: front-office/Your-vouchers.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+| 3     | Yes  | No          | No                | No                | No                        |
+| 4     | No   | No          | Yes               | No                | Yes                       |
+| 5     | No   | No          | No                | No                | Yes                       |
+| 6     | Yes  | No          | Yes               | No                | No                        |
+
+When there is no voucher for the customer, the page displays: "You do not have any vouchers." 
+
+If the customer has vouchers, it is listed on the table with the following information:
+- the voucher's Code if there is any. It leaves a blank if it is a voucher with no code.
+- The Description of the voucher, it corresponds to the name of the cart rule form.
```
Comment (LouiseBonnard) [inline front-office/Your-vouchers.md:20]:
```suggestion
- The Description of the voucher. It stands for the name of the cart rule.
```

File: front-office/Your-vouchers.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+| 2     | X    | X           | X                 | Yes               | Yes                       |
+| 3     | Yes  | No          | No                | No                | No                        |
+| 4     | No   | No          | Yes               | No                | Yes                       |
+| 5     | No   | No          | No                | No                | Yes                       |
+| 6     | Yes  | No          | Yes               | No                | No                        |
+
+When there is no voucher for the customer, the page displays: "You do not have any vouchers." 
+
+If the customer has vouchers, it is listed on the table with the following information:
+- the voucher's Code if there is any. It leaves a blank if it is a voucher with no code.
```
Comment (LouiseBonnard) [inline front-office/Your-vouchers.md:19]: Does a voucher with no code still appear in the listing?

Comment (LouiseBonnard) [reply front-office/Your-vouchers.md:19]:
```suggestion
- The voucher's Code if there is any. It leaves a blank if it is a voucher with no code.
```

File: front-office/Your-vouchers.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+On the table below are listed all the different cases.
+
+| Cases | Code | Highlighted | Group restriction | User resctriction | Is Visible Voucher page ? |
+|-------|------|-------------|-------------------|-------------------|---------------------------|
+| 1     | X    | Yes         | X                 | X                 | Yes                       |
+| 2     | X    | X           | X                 | Yes               | Yes                       |
+| 3     | Yes  | No          | No                | No                | No                        |
+| 4     | No   | No          | Yes               | No                | Yes                       |
+| 5     | No   | No          | No                | No                | Yes                       |
+| 6     | Yes  | No          | Yes               | No                | No                        |
```
Comment (LouiseBonnard) [inline front-office/Your-vouchers.md:14]: What happens when the customer uses the voucher, is it still displayed on this page?

File: front-office/Your-vouchers.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+## Rules of the displayed vouchers
+
+A voucher is displayed if it follows specific restrictions:
+
+On the table below are listed all the different cases.
+
+| Cases | Code | Highlighted | Group restriction | User resctriction | Is Visible Voucher page ? |
+|-------|------|-------------|-------------------|-------------------|---------------------------|
+| 1     | X    | Yes         | X                 | X                 | Yes                       |
```
Comment (LouiseBonnard) [inline front-office/Your-vouchers.md:9]: What is the 'X'?

Comment (MatShir) [reply front-office/Your-vouchers.md:9]: Yes and No, I could replace by Yes or No, or Yes/No

File: front-office/Your-vouchers.md (hunk 5 above)
Comment (LouiseBonnard) [reply front-office/Your-vouchers.md:14]: One question left, sorry @MatShir! ⬆️

File: front-office/Your-vouchers.md (hunk 6 above)
Comment (LouiseBonnard) [reply front-office/Your-vouchers.md:9]: It will be clearer, yes.

**Review by Robin-Fischer-PS — COMMENTED** [review]
Excepted from the already suggested change by @LouiseBonnard (replace "X" by "Yes/no"), LGTM !

File: front-office/Your-vouchers.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
+## Rules of the displayed vouchers
+
+A voucher is displayed if it follows specific restrictions:
```
Comment (marionf) [inline front-office/Your-vouchers.md:3]:
```suggestion
Vouchers are displayed in the customer account in the front-office, according to their parameters:
```

File: front-office/Your-vouchers.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+On the table below are listed all the different cases.
+
+| Cases | Code | Highlighted | Group restriction | User restriction | Available in the FO vouchers page? |
```
Comment (marionf) [inline front-office/Your-vouchers.md:7]: It depends also of the status of the cart rule, maybe we should add it

File: front-office/Your-vouchers.md (hunk 6 above)
Comment (marionf) [reply front-office/Your-vouchers.md:9]: +1 to replace the remaining "X", it's not very clear like this

File: front-office/Your-vouchers.md (hunk 5 above)
Comment (marionf) [reply front-office/Your-vouchers.md:14]: From what I tested, if the "Total available for each user" is reched or if the "Total available" is 0, the voucher is still displayed but with 0 in the quantity column
I think we should specify this in the spec, see my suggestion at the line 21

File: front-office/Your-vouchers.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+| 4     | No   | No          | Yes               | No                | Yes                       |
+| 5     | No   | No          | No                | No                | Yes                       |
+| 6     | Yes  | No          | Yes               | No                | No                        |
+
+When there is no voucher for the customer, the page displays a notification: `You do not have any vouchers.` in `Shop.Notifications.Warning`.
+
+If the customer has vouchers, it is listed on the table with the following information:
+- The voucher's Code if there is any. It leaves a blank if it is a voucher with no code.
+- The Description of the voucher. It stands for the name of the cart rule.
+- The Quantity available for the customer.
```
Comment (marionf) [inline front-office/Your-vouchers.md:21]:
```suggestion
- The Quantity available for the customer. If the "Total available for each user" is reched or if the "Total available" is 0, the voucher is still displayed but with 0 in the quantity column
```

File: front-office/Your-vouchers.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
+## Rules of the displayed vouchers
```
Comment (marionf) [inline front-office/Your-vouchers.md:1]:
```suggestion
## Rules of the vouchers displayed in the customer's account in front-office
```

**Review by marionf — CHANGES_REQUESTED** [review]
Some suggestions @MatShir

**Comment by MatShir** [comment]
@marionf je pense qu'on est bon ?

**Review by marionf — APPROVED** [review]
Yes, all good :+1:

**Comment by marionf** [comment]
Thank you @MatShir :100:

**Pull request merged**